gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java

```diff
@@ -0,0 +1,343 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.hive.metastore;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.AlreadyExistsException;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.thrift.TException;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+
+import gobblin.annotation.Alpha;
+import gobblin.configuration.State;
+import gobblin.hive.HiveMetaStoreClientFactory;
+import gobblin.hive.HivePartition;
+import gobblin.hive.HiveRegProps;
+import gobblin.hive.HiveRegister;
+import gobblin.hive.HiveRegistrationUnit.Column;
+import gobblin.hive.HiveTable;
+import gobblin.hive.HiveStripedLocks.HiveLock;
+import gobblin.hive.spec.HiveSpec;
+import gobblin.hive.spec.HiveSpecWithPostActivities;
+import gobblin.hive.spec.activity.Activity;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * An implementation of {@link HiveRegister} that uses {@link IMetaStoreClient} for Hive registration.
+ *
+ * <p>
+ *   An instance of this class is constructed with a {@link State} object or obtained via
+ *   {@link HiveRegister#get(State)}. Property {@link HiveRegProps#HIVE_DB_ROOT_DIR} is required for registering
```

Comment:
Update javadoc. Couple other references to `#register` and `HiveRegister` should be changed.